src/wallet/util_spend.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_WALLET_UTIL_SPEND_H
+#define BITCOIN_WALLET_UTIL_SPEND_H
+
+#include <policy/fees.h>
+#include <wallet/transaction.h>
+
+#include <optional>
+
+struct CreatedTransactionResult
```

Comment:
> Would suggest maybe moving this struct to wallet/types.h instead of introducing a new header. That file is meant to hold wallet types that are used outside of the wallet library. (For context see https://github.com/bitcoin/bitcoin/pull/29415#discussion_r1543379055 which talks about wallet/types.h, node/types.h, and common/types.h files).

Nice, sure. I'm on the same boat, I just wrote this without thinking on a general convention.